wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,306 @@
+from discord.ext import commands
+import discord
+from resources.utilities.embed import embed as em
+from WalleModels.models import BannedUsers, BanRecords
+import datetime
+import pytz
+import re
+from typing import Union
+import logging
+logger = logging.getLogger('wall_e')
+
+
+class Ban(commands.Cog):
+
+    def __init__(self, bot, config):
+        self.bot = bot
+        self.config = config
+        self.banlist = []
+        self.errorColour = 0xA6192E
+
+    @commands.Cog.listener(name='on_ready')
+    async def load(self):
+        """Grabs channel to send mod reports to and reads in the blacklist from db"""
+        report_channel = self.config.get_config_value('basic_config', 'COUNCIL_REPORT_CHANNEL')
+        logger.info(f'[Ban load()] Attempting to get the report channel: {report_channel}')
+
+        self.mod_channel = discord.utils.get(self.bot.guilds[0].channels, name=report_channel)
+        logger.info(f"[Ban info()] #{report_channel} channel {'successfully' if self.mod_channel else 'not'} found")
+
+        # read in blacklist of banned users
+        logger.info('[Ban load] loading ban list from the database')
+        self.banlist = await BannedUsers.get_banned_ids()
+        logger.info(f"[Ban load()] loaded the following banned users: {self.banlist}")
+
+    @commands.Cog.listener(name='on_member_join')
+    async def watchdog(self, member: discord.Member):
+        """Watches for users joining the guild and kicks and bans a user if they are banned"""
+
+        if member.id in self.banlist:
+            logger.info(f"[Ban watchdog()] banned member, {member}, detected. Promply will notify and kick them.")
+
+            e_obj = discord.Embed(title="Ban Notification",
+                                  color=discord.Color.red(),
+                                  timestamp=datetime.datetime.now(pytz.timezone('Canada/Pacific'))
+                                  )
+            e_obj.add_field(name="Notice", value=f"**You are PERMANENTLY BANNED from\n{self.bot.guilds[0]}\n\n"
+                            "You may NOT rejoin the guild!**")
+            e_obj.set_footer(icon_url=self.bot.guilds[0].icon_url, text=self.bot.guilds[0])
+
+            await member.send(embed=e_obj)
+            await member.kick(reason="Not allowed back on server.")
+
+    @commands.Cog.listener(name='on_member_ban')
+    async def intercept(self, guild: discord.Guild, member: Union[discord.User, discord.Member]):
```

Comment:
so this basically happens if a Minion accidentally triggers a guild and this convert it to a wall_e ban?

Reply:
yea. if watever reason a mod bans someone using the normal ban method this'll convert it to a wall_e ban